File: customview/src/main/java/com/chiclaim/customview/MainActivity.java
package com.chiclaim.customview;

import com.chiclaim.customview.animation.AnimationActivity;
import com.chiclaim.customview.hencoder.ui01.UI01Activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void recyclerViewNested(View view) {
        startActivity(new Intent(this, RecyclerViewInRecyclerView.class));
    }

    public void testEditTextInMeiZu(View view) {
        startActivity(new Intent(this, EditViewInMZActivity.class));
    }

    public void henCoder01(View view) {
        startActivity(new Intent(this, UI01Activity.class));
    }

    public void animation(View view){
        startActivity(new Intent(this, AnimationActivity.class));
    }


    private long currentBackPressedTime;

    @Override
    public void onBackPressed() {
        if (getSupportFragmentManager().getBackStackEntryCount() > 0) {
            getSupportFragmentManager().popBackStack();
            return;
        }
        if (System.currentTimeMillis() - currentBackPressedTime > 2000) {
            currentBackPressedTime = System.currentTimeMillis();
            Toast.makeText(this, "再按一次退出", Toast.LENGTH_SHORT).show();
        } else {
            super.onBackPressed();
        }
    }
}

File: framework/src/main/java/com/chiclaim/framework/MainActivity.kt
package com.chiclaim.framework

import android.os.Bundle
import android.os.Handler
import android.os.Message
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val start = System.currentTimeMillis()
        println("start: $start")

        val handler = Handler()
        val message = Message.obtain(handler) {
            val end = System.currentTimeMillis()
            println("end: $end")
            println("diff:" + (end - start))
        }
        handler.sendMessage(message)
    }
}